docs/features/software-catalog/life-of-an-entity.md

```diff
@@ -0,0 +1,127 @@
+
+![General overview](../../assets/features/catalog/life-of-an-entity_overview.svg)
+
+The details of these processes are described below.
+
+## Ingestion
+
+Each catalog deployment has a number of entity providers installed. They are
+responsible for fetching data from external authoritative sources in any way
+that they see fit, to translate those into entity objects, and to notify the
+database when those entities are added or removed. These are the _unprocessed
+entities_ that will be subject to later processing (see below), and they form
+the very basis of existence for entities. If there were no entity providers, no
+entities would ever enter the system.
+
+The database always keeps track of the set of entities that belong to each
+provider; no two providers can try to output the same entity. And when a
+provider signals the removal of an entity, then that leads to an _eager
+deletion_: the entity and all auxiliary data that it has led to in the database
+is immediately purged.
+
+![Ingestion overview](../../assets/features/catalog/life-of-an-entity_ingestion.svg)
+
+There are two providers installed by default: the one that deals with user
+registered locations (e.g. URLs to YAML files), and the one that deals with
+static locations in the app-config. You can add more third party providers by
+passing them to the catalog builder in your backend initialization code, and you
+can easily write your own.
+
+An entity provider is a class that implements the `EntityProvider` interface. It
+has three main parts:
+
+- The identity: Each provider instance has a unique, stable identifier that the
+  database can use to keep track of the originator of each unprocessed entity.
+- The connection: During backend startup, each provider is attached to the
+  catalog runtime.
+- The stream of events: During its lifetime, the provider can issue change
+  events to the runtime at any point in time, to modify its set of unprocessed
+  entities.
+
+It is entirely up to the provider to choose how and when it produces these
+change events. For example, the app-config provider only fires off an update at
+startup and then lies dormant. The location database provider does an initial
+update at startup, and then small delta updates every time a location database
+change is detected. The LDAP provider is driven externally by a timer loop that
+occasionally triggers a full update. Some future provider may be entirely event
+driven, feeding off an event bus or web hook. There is no magic coordination
+among providers; if they need to arrange synchronization or locking among
+themselves for example to avoid duplicate work across multiple catalog service
+machines, they need to handle that out-of-band.
+
+The entities that are emitted get some coarse validation applied to them, to
+ensure that they adhere to the most basic rules of the system. They need to have
+a `kind`, a `metadata.name`, and optionally a `metadata.namespace`. Apart from
```

Comment:
[apiVersion](https://backstage.io/docs/features/software-catalog/descriptor-format#common-to-all-kinds-the-envelope) too? Or is that not strictly required?

Reply:
Yeah that's right, the entire envelope actually has to be valid. I'll rewrite a little.